File: algs4/dataStructure/Stack.py
from .Bag import Bag

class Stack(Bag):
    '''
    这是Stack的列表实现，继承于Bag
    '''

    def __init__(self):
        super().__init__()

    def push(self, ele):
        self.add(ele)

    def pop(self):
        return self.content.pop()

File: algs4.js/dataStructure/Stack.js
/*
 * @Author: zhenbang.tang 
 * @Date: 2018-05-29 19:44:33 
 * @Last Modified by:   zhenbang.tang 
 * @Last Modified time: 2018-05-29 19:44:33 
 * 堆栈
 */

const Bag = require('./Bag');

class Stack extends Bag {
  constructor() {
    super();
  }

  push(ele) {
    this.add(ele);
  }

  pop() {
    return this.content.pop();
  }
}